app/subsystems/content/visit_book_part.rb

```
@@ -47,6 +47,8 @@ def map_visitor_names_to_visitors(visitor_names)
         outputs[:exercises] = Content::Models::ExerciseVisitor.new
       when 'page_data'
         outputs[:page_data] = Content::Models::PageDataVisitor.new
+      else
+        raise "undefined visitor #{name}. Try one of toc, exercises, page_data"
```

+1 for non-silent failure when misused.  +1 for instructive error message.